plutus-core/plutus-ir/test/transform/inline/effectfulBuiltinArg

```diff
@@ -0,0 +1,12 @@
+-- x should not be inlined, because the `trace "msg1"` call comes before
```

Comment:
I still think we should allow `x` to be inlined if `-conservative-optimisation` is off. In practice it is almost always acceptable to inline in cases like this.

Doing that separately is totally fine, although I'd imagine it is easier to implement that, and shouldn't need the `termEvaluationOrder` machinery.

Reply:
Yes, if we aren't being conservative then I think we can change the check from "is this the _first_ evaluated term" to "is this _an_ evaluated term".

Reply:
I added a note about this.